### PR TITLE
feat: Add user authentication

### PR DESCRIPTION
**Pull request by huan3846**
Introduce user sign-in with authentication tokens
Create user sign-out feature
Implement user sign-in and sign-out functionality

### CODE REVIEW

File: frontend/src/components/Navbar.tsx [hunk 1]
```diff
@@ -1,33 +1,65 @@
...
+interface NavbarProps {
+  auth: boolean;
+  username: string;
+  handleSignout: () => void;
+}
+
+function Navbar({ auth, username, handleSignout }: NavbarProps): JSX.Element {
+  
+  // If usersigned in, it displays a greeting with the username and a "Sign Out" button. 
+  // If not signed in, it displays "Sign In" and a "Sign Out" button.
```
Comment (tfurey96) [inline frontend/src/components/Navbar.tsx:17]:
```suggestion
  // If not signed in, it displays "Sign In" and a "Sign Up" button.
```

**Review by tfurey96 — CHANGES_REQUESTED** [review]

File: frontend/src/services/index.ts [hunk 2]
```diff
@@ -37,3 +37,23 @@ export async function getBookCount() {
...
+
+export async function fetchSignOut() {
+  const res = await axios.get(`${BASE_URL}/sign-out`);
+  return res;
+}
+
+export async function fetchHome() {
+  const res = await axios.get(`${BASE_URL}`);
+  return res;
+}
```
Comment (CheesyGamer77) [inline frontend/src/services/index.ts:59]: Please add documentation comments for your functions. This makes it easier for others in the team to use your code and understand what it does. An example can be found [in `backend/src/models/book.ts`](https://github.com/COSC481W-2023Fall/Bookworm/blob/main/backend/src/models/book.ts#L33-L41).

**Review by CheesyGamer77 — CHANGES_REQUESTED** [review]
A couple minor issues, but otherwise great job!

Once these issues are addressed, we can merge this. It will still say PR/lint failed, but we can fix that separately later.

**Review by tfurey96 — COMMENTED** [review]
looks good

**Review by tfurey96 — APPROVED** [review]

File: frontend/src/components/Navbar.tsx [hunk 3]
```diff
@@ -45,7 +73,7 @@ function Navbar(): JSX.Element {
           className={styles.search}
           placeholder='Book Title, Author, ISBN'
           size='large'
-          onSearch={onSearch}
+          // onSearch={onSearch}
```
[the remark this answers is not in the document]
Comment (huan3846) [reply frontend/src/components/Navbar.tsx:78]: No. Maybe I commented it by mistake

**Review by tfurey96 — APPROVED** [review]
Looks good.

**Review by CheesyGamer77 — APPROVED** [review]

**Pull request merged**